src/treeview_provider/modelTreeviewProvider.ts

```diff
@@ -104,10 +104,9 @@ abstract class ModelTreeviewProvider
     if (element) {
       return Promise.resolve(this.getTreeItems(element.key, event));
     }
-    const fileName = path.basename(
+    const fileName = path.parse(
       window.activeTextEditor!.document.fileName,
-      ".sql",
-    );
+    ).name;
```

Comment:
We should adapt this code to follow this logic for sql files, but different logic for yaml files. to use the texteditor position in case of multiple models within the yaml file, or select the single model in case of a single model in the yaml file